lms/djangoapps/grades/signals/handlers.py

```diff
@@ -56,11 +56,17 @@ def submissions_score_set_handler(sender, **kwargs):  # pylint: disable=unused-a
     course_id = kwargs['course_id']
     usage_id = kwargs['item_id']
     user = user_by_anonymous_id(kwargs['anonymous_user_id'])
+    score_deleted = False
     if user is None:
         return
     if points_possible == 0:
         # This scenario is known to not succeed, see TNL-6559 for details.
         return
+    if points_earned == 0:
+        # This can indicate either a cancellation or an actual 0/x score.
+        # The former case requires score_deleted=True and the latter isn't harmed by it
+        # See TNL-6679 for details
+        score_deleted = True
```

Comment:
This logic is a holdover from a previous attempt to allow the set signal, but handle it properly in a clear-student-state operation. I don't believe it's needed under the new approach.